packages/application-generic/src/modules/queues.module.ts

```diff
@@ -67,3 +67,19 @@ const PROVIDERS: Provider[] = [
   exports: [...PROVIDERS],
 })
 export class QueuesModule {}
+
+const APP_PROVIDERS: Provider[] = [
+  InboundParseQueue,
+  InboundParseWorker,
+  InboundParseQueueServiceHealthIndicator,
+  WebSocketsQueueService,
+  WebSocketsQueueServiceHealthIndicator,
+  WorkflowQueueService,
+  WorkflowQueueServiceHealthIndicator,
+];
+
+@Module({
+  providers: [...APP_PROVIDERS],
+  exports: [...APP_PROVIDERS],
+})
+export class BaseApiQueuesModule {}
```

Comment:
Are you trying to gather all the QueuesModule dependencies used only in API app?

Reply:
Yes exactly, for two reasons 
1. that way we won't initialize redundant clients in API.
2. in the near future (i hope) i want to try creating dynamic health validation, meaning if we inject a new HealthIndicator in the DI it will be validated out of the box, for example without adding it to the healthCheck in the api/src/app/health/health.controller.ts.

Reply:
From the small test I made locally before making the changes I saw that they do initialize even if not injected, i guess because they are singletons maybe?

Regarding the separation i totally agree, make it only for the API in order to keep the scope as small as possible for this PR as it is an extra for it. We definitely can update the other ones as well.

Once again I agree we can merge this one for now, the additional improvements can wait for now. I made this one as a preparation for the future.